src/common/utils/ldap/ldap.go

```diff
@@ -0,0 +1,367 @@
+/*
+   Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ldap
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"crypto/tls"
+
+	"github.com/vmware/harbor/src/common/dao"
+	"github.com/vmware/harbor/src/common/models"
+	"github.com/vmware/harbor/src/common/utils/log"
+	"github.com/vmware/harbor/src/ui/config"
+
+	goldap "gopkg.in/ldap.v2"
+)
+
+var attributes = []string{"uid", "cn", "mail", "email"}
+
+// GetSystemLdapConf ...
+func GetSystemLdapConf() (models.LdapConf, error) {
+	var err error
+	var ldapConfs models.LdapConf
+
+	if config.AuthMode() != "ldap_auth" {
```

Comment:
from original code: /src/ui/config/config.go 43~56
I think we are consistent.
`	mode := raw["AUTH_MODE"]`
`	if mode == "ldap_auth" {`
`		setting := LDAPSetting{`
`			URL:            raw["LDAP_URL"],`
`			BaseDn:         raw["LDAP_BASE_DN"],`
`			SearchDn:       raw["LDAP_SEARCH_DN"],`
`			SearchPwd:      raw["LDAP_SEARCH_PWD"],`
`			UID:            raw["LDAP_UID"],`
`			Filter:         raw["LDAP_FILTER"],`
`			Scope:          raw["LDAP_SCOPE"],`
`			ConnectTimeout: raw["LDAP_CONNECT_TIMEOUT"],`
`		}`
		config["ldap"] = setting
	}`
Otherwise need add same code in 4 parts of code:
ui/api/ldap.go
  Ping()
  Search()
  ImportUser()
ui/auth/ldap.go
  Authenticate()